### ISSUE TITLE
Pending Fedora 35 package does not build (new TeX Live?)

### ISSUE THREAD

**Issue by flyn-org** [brucemiller/LaTeXML#1565]
I maintain the LaTeXML package for Fedora. The package just stopped building on (pre-release) Fedora 35, possibly due to the upgrade of TeX Live there. More information is available at https://bugzilla.redhat.com/show_bug.cgi?id=1969480.

Has anyone had success running the LaTeXML tests on TeX Live svn57862 or newer?

**Comment by dginev** [comment]
Is the implication from Fedora that we need to keep a successfully building CI setup against the bleeding-edge of TeXlive?

We don't test specific svn variants, just the annual releases, and we tend to be quite slow/conservative with updating to newer ones at the moment.

expl3 regressing is indeed indicative that the newest texlive may have moved beyond latexml's  currently implemented support for LaTeX 3.

**Comment by flyn-org** [comment]
@dginev, I am not sure a successfully-building CI setup against the bleeding-edge of TeX Live is necessary, although it might be helpful. I am the maintainer, so it is my responsibility to alert you if something goes awry as it did here. Packagers such as I prefer fixing issues upstream, but sometimes we patch things to get them to work temporarily. As you probably know, Fedora is more aggressive than some other distributions at pursuing the latest packages, so this kind of thing happens.

**Comment by dginev** [comment]
Got it, thanks for letting us know right away @flyn-org , much appreciated! And of course thanks for continuing to maintain the package, that's super helpful.

**Comment by dginev** [comment]
Actually @flyn-org now that I searched for for the messages properly, this has been patched in #1454  and should work reliably in the latest latexml master. So... Would the preferred solution on our end be to ship a new release of latexml? Unsure how to best tackle that.

**Comment by flyn-org** [comment]
I am always happy to package a new version, if you are ready for that. Otherwise, I can backport the patches to allow things to work on the forthcoming release of Fedora. I will do the latter now, and I will package a new release if you make one.

**Comment by flyn-org** [comment]
I was able to backport https://github.com/brucemiller/LaTeXML/pull/1547. This fixed some of the errors, but the following remain:

Error:unexpected:undef Unexpected input remaining: 'undef'
	at xparse.sty.ltxml; line 19
	Finished reading from Mouth[/usr/share/texlive/texmf-dist/tex/latex/l3packages/xparse/xparse.sty@0x0], but it still has input.
	Input is empty
	In Core::Gullet[@0x2aa11047448] /builddir/build/BUILD/LaTeXML-0.8.5/blib/lib/LaTeXML/Package/xparse.sty.ltxml; line 19
	 <= Core::Definition::Constructor[\usepac... <= Core::Stomach[@0x2aa10ef9978] <= ...
Error:unexpected:T_CS[\NewDocumentCommand] Unexpected input remaining: 'T_CS[\NewDocumentCommand]'
	at xparse.tex; line 4 col 19 - line 4 col 19
	Finished reading from Mouth[/usr/share/texlive/texmf-dist/tex/latex/l3packages/xparse/xparse.sty@0x0], but it still has input.
	Next token is T_BEGIN[{]
	In Core::Gullet[@0x2aa11047448] /builddir/build/BUILD/LaTeXML-0.8.5/t/expl3/xparse.tex; from line 4 col 21 to line 4 col 21
	 <= Core::Definition::Constructor[\usepac... <= Core::Stomach[@0x2aa10ef9978] <= ...
Error:unexpected:<closed> Mouth is unexpectedly already closed
	at String; line 0 col 0 - line 0 col 0
	Reading from Mouth[/usr/share/texlive/texmf-dist/tex/latex/l3packages/xparse/xparse.sty@0x0], but it has already been closed.
	Input is empty
	In Core::Gullet[@0x2aa11047448] Anonymous String; from line 1 col 0 to line 1 col 0
	 <= Core::Definition::Constructor[\usepac... <= Core::Stomach[@0x2aa10ef9978] <= ...
Error:unexpected:<closed> Mouth is unexpectedly already closed
	at String; line 1 col 0 - line 1 col 0
	Reading from Mouth::Binding[/builddir/build/BUILD/LaTeXML-0.8.5/blib/lib/LaTeXML/Package/xparse.sty.ltxml], but it has already been closed.
	Input is empty
	In Core::Gullet[@0x2aa11047448] Anonymous String; from line 2 col 0 to line 2 col 0
	 <= Core::Definition::Constructor[\usepac... <= Core::Stomach[@0x2aa10ef9978] <= ...

**Comment by dginev** [comment]
Thanks for following up @flyn-org ! 

The mention of `\NewDocumentCommand` in xparse points to our currently open issue #1566 and the suggest patch at #1569 . Thanks for letting us know this is a remaining issue to get the Fedora 35 package working, I think we should give the PR extra priority in that case, so that we get to _some_ resolution. Will alert @brucemiller about that as well.

**Comment by flyn-org** [comment]
I added the patch at https://github.com/brucemiller/LaTeXML/pull/1569 to https://github.com/brucemiller/LaTeXML/pull/1547, but I still received the error above when building.

**Comment by flyn-org** [comment]
I was able to build on Fedora Rawhide (which will become Fedora 35) with the patch from https://github.com/brucemiller/LaTeXML/pull/1578. Please see https://koji.fedoraproject.org/koji/taskinfo?taskID=72544751 for the result. So, I built a patched 0.8.5, and I will replace this with the next release once it is available. I will leave this issue open until the release.

Thank you @brucemiller and @dginev!

**Comment by xworld21** [comment]
@flyn-org would you mind sharing which commits you decided to backport to Fedora? I am trying to do the same for nixpkgs (I cherry picked https://github.com/brucemiller/LaTeXML/commit/be203639aa7674b76cf17fe5dae284385b4f434c, https://github.com/brucemiller/LaTeXML/commit/efb5776080bc13f5e78d3b86a04ebf02a542e42a, https://github.com/brucemiller/LaTeXML/commit/b034118d0df281212e0abdab2c6e4ef9775b43eb) but I am still getting errors. What else needs to go in?

**Comment by flyn-org** [comment]
I went the lazy route and pulled everything from master. I hope to package the next release soon, if it materializes.

**Comment by xworld21** [comment]
> I went the lazy route and pulled everything from master. I hope to package the next release soon, if it materializes.

Right. It sounds like I'll have to do the same, the code has diverged too much from 0.8.5 for a simple backport.

**Comment by dginev** [comment]
> > I went the lazy route and pulled everything from master. I hope to package the next release soon, if it materializes.
> 
> Right. It sounds like I'll have to do the same, the code has diverged too much from 0.8.5 for a simple backport.

Just to remark that this is quite appropriate, as we've been discussing shipping 0.8.6 quite soon - hopefully end of August.
